File: easy/go/0002.go
func twoSum(nums []int, target int) []int {
    table := make(map[int]int)
    ans := make([]int, 2)
    for i := 0; i < len(nums); i++{
        if(table[target - nums[i]] != 0){
            ans[0] = i
            ans[1] = table[target - nums[i]] - 1
            return ans
        }else{
            table[nums[i]] = i+1
        }
    }
    return ans
}

File: easy/cpp/0002.cpp
#include <vector>
#include <unordered_map>
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
    	unordered_map<int, int> table;
        vector<int> ans;
    	for(int i = 0 ; i < nums.size(); i++){
    		if(table[target - nums[i]]){
                ans.push_back(i);
                ans.push_back(table[target - nums[i]]-1);
    			break;
    		}else{
    			table[nums[i]] = i+1;
    		}
    	}
        return ans;
    }
};